src/main.cpp

```diff
@@ -600,13 +639,18 @@ int main(int argc, char**argv)
   }
 
   realm_manager->stop();
+
   delete realm_manager; realm_manager = NULL;
   delete diameter_resolver; diameter_resolver = NULL;
   delete http_resolver; http_resolver = NULL;
   delete dns_resolver; dns_resolver = NULL;
-
   delete load_monitor; load_monitor = NULL;
 
+  hc->terminate();
+  pthread_join(health_check_thread, NULL);
+  delete hc; hc = NULL;
+  delete exception_handler; exception_handler = NULL;
```

Comment:
Should probably reverse these and delete the health-checker after the exception handler. (Sorry, that might be my fault.)